com.avaloq.tools.ddk.xtext.generator/src/com/avaloq/tools/ddk/xtext/generator/parser/antlr/CombinedGrammarReportBuilder.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.generator.parser.antlr;
+
+import java.util.List;
+
+import org.eclipse.xtext.AbstractRule;
+import org.eclipse.xtext.EnumRule;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.GrammarUtil;
+import org.eclipse.xtext.ParserRule;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.xtext.RuleWithParameterValues;
+import org.eclipse.xtext.xtext.generator.grammarAccess.GrammarAccessExtensions;
+
+
+/**
+ * Helper class to generate combined grammar as HTML.
+ */
+@SuppressWarnings("nls")
+public class CombinedGrammarReportBuilder {
+
+  private final GrammarAccessExtensions grammarExtensions;
+
+  public CombinedGrammarReportBuilder(final GrammarAccessExtensions grammarExtensions) {
+    this.grammarExtensions = grammarExtensions;
+  }
+
+  /**
+   * Simple Xtext Doc. With analysis results

```

Comment:
Is the return value not an HTML document, instead of a Xtext one?

Reply:
I do not think I should change this one. I think the intent is to get documentation for Xtext file. 
Please refer to KeywordAnalysisHelper.printReport().
I hope you agree with me.